go/store/types/noms_kind.go

```diff
@@ -52,52 +52,55 @@ const (
 	UintKind
 	NullKind
 	TupleKind
+	UnderlyingArrayKind
```

Comment:
Probably want to call this a ByteArray, but let's see what Aaron thinks.

Reply:
That's totally fine, I wasn't too sure about the name anyway lol